### PR TITLE
chore: pin linter dependencies

### PR DESCRIPTION
**Pull request by josecorella**
*Description of changes:*
Pins linter dependencies for maintenance branch 3.x

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -223,27 +231,36 @@ commands =
         test/integration/ \
         {posargs}
 
+[testenv:pylint-examples]
+basepython = {[testenv:pylint]basepython}
+deps = {[testenv:pylint]deps}
+commands =
+    pylint --rcfile=examples/src/pylintrc examples/src/aws_encryption_sdk_cli_examples/
+    pylint --rcfile=examples/test/pylintrc --disable R0801 examples/test/
```
Comment (alex-chew) [inline tox.ini:239]: nit: any reason not to disable R0801 directly in the pylintrc?

**Review by alex-chew — DISMISSED** [review]

File: tox.ini (hunk 1 above)
Comment (josecorella) [reply tox.ini:239]: i copy pasta'd what we did for the python esdk: https://github.com/aws/aws-encryption-sdk-python/blob/master/tox.ini#L167

Comment (josecorella) [reply tox.ini:239]: I'll go ahead and move it

Comment (josecorella) [reply tox.ini:239]: plus we are already disabling it in the pylintrc: https://github.com/aws/aws-encryption-sdk-cli/blob/28051dc6df9be97de2d3b2072ad7d4698ffec241/examples/test/pylintrc#L9

**Review by alex-chew — APPROVED** [review]
LGTM

**Pull request merged**